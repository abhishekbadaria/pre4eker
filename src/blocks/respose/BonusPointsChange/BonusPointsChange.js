import React from "react";
import PropTypes from "prop-types";
import stylesShared from "src/customer.css";

const BonusPointsChange = ({ bonusPointsChanges }) => {
  const {
    balanceType,
    earnedAmount,
    spentAmount,
  } = bonusPointsChanges;

  return (
    <div className={stylesShared.customer}>
      <div className={stylesShared.inline}>
        <div className={stylesShared.half}>
          <div className={stylesShared.promoType}>Изменения баллов</div>
        </div>

        <div className={stylesShared.half}>
          <div className={stylesShared.name}>Балльный счет</div>
          <div className={stylesShared.value}>{balanceType.name}</div>
          <div className={stylesShared.status}>{balanceType.ids.systemName}</div>
        </div>
      </div>
      <div className={stylesShared.inline}>
        <div className={stylesShared.half}>
          <div className={stylesShared.name}>
          Начислено
          </div>
          <div className={stylesShared.promoValue}>
            {earnedAmount || "–"}
          </div>
        </div>
      <div className={stylesShared.half}>
      <div className={stylesShared.name}>
        Списано 
          </div>
          <div className={stylesShared.half}>
            <div className={stylesShared.promoValue}>
            {spentAmount || "–"}
            </div>
          </div>
        </div>
      </div> 
    </div>
  );
};

BonusPointsChange.propTypes = {};

export default BonusPointsChange;
