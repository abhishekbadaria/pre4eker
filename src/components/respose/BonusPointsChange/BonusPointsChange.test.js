import React from "react";
import { render, cleanup } from "@testing-library/react";
import BonusPointsChange from "./BonusPointsChange";

afterEach(cleanup);

const bonusChange = [
  {
    balanceType: {
      ids: {
        systemName: "MBcRZtuf",
      },
      name: "P02zXwLm",
    },
    earnedAmount: 1396638.62,
    spentAmount: 5761799.72,
  },
  {
    balanceType: {
      ids: {
        systemName: "REDNhFOU",
      },
      name: "LE0pZEmG",
    },
    earnedAmount: 20955125.72,
    spentAmount: 10868072.91,
  },
];

test("bonus point change match snapshot", () => {
  const { asFragment } = render(
    <BonusPointsChange bonusPointsChanges={bonusChange} />
  );
  expect(asFragment(<BonusPointsChange />)).toMatchSnapshot();
});
