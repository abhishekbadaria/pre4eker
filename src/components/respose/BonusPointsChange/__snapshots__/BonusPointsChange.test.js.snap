// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bonus point change match snapshot 1`] = `
<DocumentFragment>
  <div
    class="promo"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Изменения баллов
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="third"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          P02zXwLm
        </div>
        <div
          class="status neutral"
        >
          MBcRZtuf
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="value"
        >
          1 396 638,62 ₽
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="value"
          >
            5 761 799,72 ₽
          </div>
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="third"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          LE0pZEmG
        </div>
        <div
          class="status neutral"
        >
          REDNhFOU
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="value"
        >
          20 955 125,72 ₽
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="value"
          >
            10 868 072,91 ₽
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
